okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -651,16 +648,24 @@ private void fireTimer(Tenant tenant, ModuleDescriptor md, RoutingEntry re, Stri
   }
 
   private Future<Void> invokePermissionsForModule(Tenant tenant, ModuleDescriptor mdTo,
-                                                  ModuleDescriptor permsModule, ProxyContext pc) {
+      ModuleDescriptor mdFrom, ModuleDescriptor permsModule, ProxyContext pc) {
 
     logger.debug("Loading permissions for {} (using {})", mdTo.getName(), permsModule.getName());
     String moduleTo = mdTo.getId();
     PermissionList pl = null;
     InterfaceDescriptor permInt = permsModule.getSystemInterface("_tenantPermissions");
-    if (permInt.getVersion().equals("1.0")) {
-      pl = new PermissionList(moduleTo, mdTo.getPermissionSets());
+    String permIntVer = permInt.getVersion();
+    if (permIntVer.equals("1.0")) {
+      pl = new PermissionListV1(moduleTo, mdTo.getPermissionSets());
+    } else if (permIntVer.equals("1.1")) {
+      pl = new PermissionListV1(moduleTo, mdTo.getExpandedPermissionSets());
     } else {
-      pl = new PermissionList(moduleTo, mdTo.getExpandedPermissionSets());
+      if (mdFrom != null) {
+        pl = new PermissionListV2(moduleTo, mdFrom.getId(), mdTo.getExpandedPermissionSets(),
+            mdFrom.getExpandedPermissionSets());
+      } else {
+        pl = new PermissionListV2(moduleTo, mdTo.getExpandedPermissionSets());
+      }
```

Comment:
Please add check for "2.0". And report failure if no match (return failed future).

Reply:
And in the case where we get something other than 1.0, 1.1, or 2.0, also return a failure?  I guess the assumption there is that when/if 2.1 comes along we'll need to make changes here anyway?

Reply:
NM I need to read more closely.

Reply:
Yes. Okapi should report a failure for an unknown interface version.